src/main/java/io/gravitee/policy/cache/invoker/CacheInvoker.java

```diff
@@ -123,7 +123,11 @@ public Completable invoke(ExecutionContext executionContext) {
                                 .getHeaders()
                                 .forEach((key, values) -> values.forEach(value -> response.headers().add(key, value)));
                         }
-                        return response.onBody(body -> body.ignoreElement().andThen(Maybe.just(cacheResponse.getContent())));
+                        return response.onBody(body ->
+                            body
+                                .ignoreElement()
+                                .andThen(Maybe.just(Buffer.buffer(Base64.getDecoder().decode(cacheResponse.getContent().getBytes()))))
```

Comment:
We have discussed it many times and agree that this deserves a complete refactoring to allow direct use of POJO when the in-memory cache is used. That being said, base64 encode/decode is a costly operation that comes in addition to a JSON serialization which starts becoming very CPU-consuming. I'm wondering if we could rely on HTTP header Content-Type to encoding/decoding for the well-known text-based content (text/plain , application/xml, application/json), or do the opposite: encode/decode when application/octet-stream?